

<!DOCTYPE html>
<html lang="zh-CN" data-default-color-scheme=auto>



<head>
  <meta charset="UTF-8">
  <link rel="apple-touch-icon" sizes="76x76" href="/img/favicon.png">
  <link rel="icon" href="/img/favicon.png">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=5.0, shrink-to-fit=no">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  
  <meta name="theme-color" content="#2f4154">
  <meta name="author" content="John Doe">
  <meta name="keywords" content="">
  
    <meta name="description" content="猕猴胎脑CP层ChIP-seq分析(上)上游处理：0、需要的软件及需要提前建的目录12345678910#需要的软件#下载：sra-tools#质控：fastqc、multiqc、trim-galore#比对：bowtie2#call peaks：MACS2#可视化：deeptools、igv#motif寻找：homer#需要提前建的目录align  clean  motif  peaks  qc">
<meta property="og:type" content="article">
<meta property="og:title" content="猕猴胎脑CP层ChIP-seq分析(上)">
<meta property="og:url" content="https://caigouchen.github.io/2022/02/10/%E7%8C%95%E7%8C%B4%E8%83%8E%E8%84%91CP%E5%B1%82ChIP-seq%E5%88%86%E6%9E%90-%E4%B8%8A/index.html">
<meta property="og:site_name" content="caigouchen">
<meta property="og:description" content="猕猴胎脑CP层ChIP-seq分析(上)上游处理：0、需要的软件及需要提前建的目录12345678910#需要的软件#下载：sra-tools#质控：fastqc、multiqc、trim-galore#比对：bowtie2#call peaks：MACS2#可视化：deeptools、igv#motif寻找：homer#需要提前建的目录align  clean  motif  peaks  qc">
<meta property="og:locale" content="zh_CN">
<meta property="og:image" content="https://caigouchen.github.io/2022/02/10/%E7%8C%95%E7%8C%B4%E8%83%8E%E8%84%91CP%E5%B1%82ChIP-seq%E5%88%86%E6%9E%90-%E4%B8%8A/image-20220208143628790.png">
<meta property="og:image" content="https://caigouchen.github.io/2022/02/10/%E7%8C%95%E7%8C%B4%E8%83%8E%E8%84%91CP%E5%B1%82ChIP-seq%E5%88%86%E6%9E%90-%E4%B8%8A/image-20220210201437081.png">
<meta property="og:image" content="https://caigouchen.github.io/2022/02/10/%E7%8C%95%E7%8C%B4%E8%83%8E%E8%84%91CP%E5%B1%82ChIP-seq%E5%88%86%E6%9E%90-%E4%B8%8A/image-20220210220849746.png">
<meta property="og:image" content="https://caigouchen.github.io/2022/02/10/%E7%8C%95%E7%8C%B4%E8%83%8E%E8%84%91CP%E5%B1%82ChIP-seq%E5%88%86%E6%9E%90-%E4%B8%8A/image-20220210221139367.png">
<meta property="article:published_time" content="2022-02-10T14:15:21.000Z">
<meta property="article:modified_time" content="2022-02-10T14:23:21.413Z">
<meta property="article:author" content="John Doe">
<meta property="article:tag" content="组学分析">
<meta property="article:tag" content="ChIP-seq">
<meta name="twitter:card" content="summary_large_image">
<meta name="twitter:image" content="https://caigouchen.github.io/2022/02/10/%E7%8C%95%E7%8C%B4%E8%83%8E%E8%84%91CP%E5%B1%82ChIP-seq%E5%88%86%E6%9E%90-%E4%B8%8A/image-20220208143628790.png">
  
  
  <title>猕猴胎脑CP层ChIP-seq分析(上) - caigouchen</title>

  <link  rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap@4/dist/css/bootstrap.min.css" />


  <link  rel="stylesheet" href="https://cdn.jsdelivr.net/npm/github-markdown-css@4/github-markdown.min.css" />
  <link  rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hint.css@2/hint.min.css" />

  
    
    
      
      <link  rel="stylesheet" href="https://cdn.jsdelivr.net/npm/highlight.js@10/styles/github-gist.min.css" />
    
  

  
    <link  rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@3/dist/jquery.fancybox.min.css" />
  


<!-- 主题依赖的图标库，不要自行修改 -->

<link rel="stylesheet" href="//at.alicdn.com/t/font_1749284_ba1fz6golrf.css">



<link rel="stylesheet" href="//at.alicdn.com/t/font_1736178_lbnruvf0jn.css">


<link  rel="stylesheet" href="/css/main.css" />

<!-- 自定义样式保持在最底部 -->


  <script id="fluid-configs">
    var Fluid = window.Fluid || {};
    var CONFIG = {"hostname":"caigouchen.github.io","root":"/","version":"1.8.14","typing":{"enable":true,"typeSpeed":70,"cursorChar":"_","loop":false},"anchorjs":{"enable":true,"element":"h1,h2,h3,h4,h5,h6","placement":"right","visible":"hover","icon":""},"progressbar":{"enable":true,"height_px":3,"color":"#29d","options":{"showSpinner":false,"trickleSpeed":100}},"copy_btn":true,"image_zoom":{"enable":true,"img_url_replace":["",""]},"toc":{"enable":true,"headingSelector":"h1,h2,h3,h4,h5,h6","collapseDepth":0},"lazyload":{"enable":true,"loading_img":"/img/loading.gif","onlypost":false,"offset_factor":2},"web_analytics":{"enable":true,"baidu":null,"google":null,"gtag":null,"tencent":{"sid":null,"cid":null},"woyaola":null,"cnzz":null,"leancloud":{"app_id":"3JBe3zLxJxVseKR7pS2aH2i1-gzGzoHsz","app_key":"Fhs2HRHSer5K9ydrUT7bBHXF","server_url":null,"path":"window.location.pathname","ignore_local":false}},"search_path":"/local-search.xml"};
  </script>
  <script  src="/js/utils.js" ></script>
  <script  src="/js/color-schema.js" ></script>
<meta name="generator" content="Hexo 5.4.0"></head>


<body>
  <header style="height: 70vh;">
    <nav id="navbar" class="navbar fixed-top  navbar-expand-lg navbar-dark scrolling-navbar">
  <div class="container">
    <a class="navbar-brand" href="/">
      <strong>CaigouChen&#39;s Blog</strong>
    </a>

    <button id="navbar-toggler-btn" class="navbar-toggler" type="button" data-toggle="collapse"
            data-target="#navbarSupportedContent"
            aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
      <div class="animated-icon"><span></span><span></span><span></span></div>
    </button>

    <!-- Collapsible content -->
    <div class="collapse navbar-collapse" id="navbarSupportedContent">
      <ul class="navbar-nav ml-auto text-center">
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/">
                <i class="iconfont icon-home-fill"></i>
                首页
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/archives/">
                <i class="iconfont icon-archive-fill"></i>
                归档
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/categories/">
                <i class="iconfont icon-category-fill"></i>
                分类
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/tags/">
                <i class="iconfont icon-tags-fill"></i>
                标签
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/about/">
                <i class="iconfont icon-user-fill"></i>
                关于
              </a>
            </li>
          
        
        
          <li class="nav-item" id="search-btn">
            <a class="nav-link" target="_self" href="javascript:;" data-toggle="modal" data-target="#modalSearch" aria-label="Search">
              &nbsp;<i class="iconfont icon-search"></i>&nbsp;
            </a>
          </li>
        
        
          <li class="nav-item" id="color-toggle-btn">
            <a class="nav-link" target="_self" href="javascript:;" aria-label="Color Toggle">&nbsp;<i
                class="iconfont icon-dark" id="color-toggle-icon"></i>&nbsp;</a>
          </li>
        
      </ul>
    </div>
  </div>
</nav>

    <div class="banner" id="banner" parallax=true
         style="background: url('/img/default.png') no-repeat center center;
           background-size: cover;">
      <div class="full-bg-img">
        <div class="mask flex-center" style="background-color: rgba(0, 0, 0, 0.3)">
          <div class="page-header text-center fade-in-up">
            <span class="h2" id="subtitle" title="猕猴胎脑CP层ChIP-seq分析(上)">
              
            </span>

            
              <div class="mt-3">
  
  
    <span class="post-meta">
      <i class="iconfont icon-date-fill" aria-hidden="true"></i>
      <time datetime="2022-02-10 22:15" pubdate>
        2022年2月10日 晚上
      </time>
    </span>
  
</div>

<div class="mt-1">
  
    <span class="post-meta mr-2">
      <i class="iconfont icon-chart"></i>
      7.3k 字
    </span>
  

  
    <span class="post-meta mr-2">
      <i class="iconfont icon-clock-fill"></i>
      
      
      61 分钟
    </span>
  

  
  
    
      <!-- LeanCloud 统计文章PV -->
      <span id="leancloud-page-views-container" class="post-meta" style="display: none">
        <i class="iconfont icon-eye" aria-hidden="true"></i>
        <span id="leancloud-page-views"></span> 次
      </span>
    
  
</div>

            
          </div>

          
        </div>
      </div>
    </div>
  </header>

  <main>
    
      

<div class="container-fluid nopadding-x">
  <div class="row nomargin-x">
    <div class="d-none d-lg-block col-lg-2"></div>
    <div class="col-lg-8 nopadding-x-md">
      <div class="container nopadding-x-md" id="board-ctn">
        <div class="py-5" id="board">
          <article class="post-content mx-auto">
            <!-- SEO header -->
            <h1 style="display: none">猕猴胎脑CP层ChIP-seq分析(上)</h1>
            
            <div class="markdown-body">
              <h2 id="猕猴胎脑CP层ChIP-seq分析-上"><a href="#猕猴胎脑CP层ChIP-seq分析-上" class="headerlink" title="猕猴胎脑CP层ChIP-seq分析(上)"></a>猕猴胎脑CP层ChIP-seq分析(上)</h2><h3 id="上游处理："><a href="#上游处理：" class="headerlink" title="上游处理："></a>上游处理：</h3><h4 id="0、需要的软件及需要提前建的目录"><a href="#0、需要的软件及需要提前建的目录" class="headerlink" title="0、需要的软件及需要提前建的目录"></a>0、需要的软件及需要提前建的目录</h4><figure class="highlight python"><table><tr><td class="gutter"><div class="code-wrapper"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></div></td><td class="code"><pre><code class="hljs python"><span class="hljs-comment">#需要的软件</span><br><span class="hljs-comment">#下载：sra-tools</span><br><span class="hljs-comment">#质控：fastqc、multiqc、trim-galore</span><br><span class="hljs-comment">#比对：bowtie2</span><br><span class="hljs-comment">#call peaks：MACS2</span><br><span class="hljs-comment">#可视化：deeptools、igv</span><br><span class="hljs-comment">#motif寻找：homer</span><br><span class="hljs-comment">#需要提前建的目录</span><br>align  clean  motif  peaks  qc  qc_after  raw  sra  TSS<br><span class="hljs-comment">#数据来源：3D Genome of macaque fetal brain reveals evolutionary innovations during primate corticogenesis中猕猴CP层CTCF-chipseq数据</span><br></code></pre></td></tr></table></figure>
<h4 id="1、下载数据"><a href="#1、下载数据" class="headerlink" title="1、下载数据"></a>1、下载数据</h4><h5 id="wget下载数据"><a href="#wget下载数据" class="headerlink" title="wget下载数据"></a>wget下载数据</h5><figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><code class="hljs python"><span class="hljs-comment">#在写这篇博客时，prefetch下载数据不好使了</span><br>SRR13252796<span class="hljs-comment">#CP-chip-seq</span><br>SRR13252798<span class="hljs-comment">#CTCF-chip-seq input</span><br><span class="hljs-comment">##############################################################################</span><br><span class="hljs-comment">#进入base环境</span><br>source ~/.bashrc &amp;&amp; conda activate chipseq<br>cd /sde/ckm/ChIP-seq/RM-CTCF<br>wget -c https://sra-downloadb.be-md.ncbi.nlm.nih.gov/sos3/sra-pub-run-<span class="hljs-number">20</span>/SRR13252796/SRR13252796<span class="hljs-number">.1</span><br>wget -c https://sra-downloadb.be-md.ncbi.nlm.nih.gov/sos3/sra-pub-run-<span class="hljs-number">21</span>/SRR13252798/SRR13252798<span class="hljs-number">.1</span><br><span class="hljs-comment">#将名字改成真实样品名称</span><br>mv SRR13252796<span class="hljs-number">.1</span> RM_CP_CTCF-ChIP-seq<br>mv SRR13252798<span class="hljs-number">.1</span> RM_CP_CTCF-ChIP-seq_input<br></code></pre></td></tr></table></figure>
<h5 id="解压缩"><a href="#解压缩" class="headerlink" title="解压缩"></a>解压缩</h5><figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><code class="hljs python"><span class="hljs-comment">#fastq-dump转换为fastq文件：</span><br><span class="hljs-comment">#返回值4为单端测序，8位双端</span><br>fastq-dump -X <span class="hljs-number">1</span> --split-spot -Z RM_CP_CTCF-ChIP-seq | wc -l<br><span class="hljs-comment">#fastq-dump具体参数</span><br><span class="hljs-comment">#--split-3双端单端都可，-O指定生成路径，最后加样品名称</span><br>cd /sde/ckm/ChIP-seq/RM-CTCF/sra<br>nohup fastq-dump --split-<span class="hljs-number">3</span> --gzip * -O ../raw &amp;<br></code></pre></td></tr></table></figure>
<h4 id="2、数据质控qc"><a href="#2、数据质控qc" class="headerlink" title="2、数据质控qc"></a>2、数据质控qc</h4><figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><code class="hljs python">cd /sde/ckm/ChIP-seq/RM-CTCF/raw<br>ls *.gz |<span class="hljs-keyword">while</span> read <span class="hljs-built_in">id</span>; do (nohup fastqc -t <span class="hljs-number">10</span> -o ../qc $<span class="hljs-built_in">id</span> &amp; );done<br><span class="hljs-comment">###############fastqc主要参数</span><br>fastqc [-o outputdir] [--(no)extract] [-f fastq|bam|sam]  seqfile1-N<br><span class="hljs-comment"># 主要是包括前面的各种选项和最后面的可以加入N个文件</span><br><span class="hljs-comment"># -o：--outdir FastQC生成的报告文件的储存路径，生成的报告的文件名是根据输入来定的</span><br><span class="hljs-comment"># --extract： 生成的报告默认会打包成1个压缩文件，使用这个参数是让程序不打包</span><br><span class="hljs-comment"># -t：--threads 选择程序运行的线程数，每个线程会占用250MB内存，越多越快咯</span><br><span class="hljs-comment"># -c：--contaminants 污染物选项，输入的是一个文件，格式是Name [Tab] Sequence，里面是可能的污染序列，如果有这个选项，FastQC会在计算时候评估污染的情况，并在统计的时候进行分析，一般用不到</span><br><span class="hljs-comment"># -a：--adapters 也是输入一个文件，文件的格式Name [Tab] Sequence，储存的是测序的adpater序列信息，如果不输入，目前版本的FastQC就按照通用引物来评估序列时候有adapter的残留</span><br><span class="hljs-comment"># -q：--quiet 安静运行模式，一般不选这个选项的时候，程序会实时报告运行的状况。</span><br><br><span class="hljs-comment">#multiqc整合结果</span><br>multiqc ./<br><span class="hljs-comment">#看结果这个数据质量一般，所以过滤掉低质量的reads很重要</span><br></code></pre></td></tr></table></figure>
<p><img src="image-20220208143628790.png" srcset="/img/loading.gif" lazyload></p>
<h4 id="3、数据过滤"><a href="#3、数据过滤" class="headerlink" title="3、数据过滤"></a>3、数据过滤</h4><h5 id="trim-galore"><a href="#trim-galore" class="headerlink" title="trim-galore"></a>trim-galore</h5><figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><code class="hljs python">cd /sde/ckm/ChIP-seq/RM-CTCF/raw<br><span class="hljs-comment">#trim-galore</span><br>ls *gz |<span class="hljs-keyword">while</span> read <span class="hljs-built_in">id</span>;do (nohup trim_galore -q <span class="hljs-number">20</span> --phred33 --length <span class="hljs-number">20</span> -e <span class="hljs-number">0.1</span> --stringency <span class="hljs-number">4</span> -o ../clean $<span class="hljs-built_in">id</span> &amp;);done<br><span class="hljs-comment">###############trim_galore主要参数</span><br><span class="hljs-comment"># --phred33：指示 Cutadapt 使用 ASCII+33 质量分数作为 Phred 分数（Sanger/Illumina 1.9+ 编码）进行质量修整。</span><br><span class="hljs-comment"># --stringency &lt;INT&gt;：修剪序列所需的adapter序列重叠。默认为非常严格的设置1，即使是重叠序列的单个碱基对，也会从任何读取的3&#x27;端修剪掉。</span><br><span class="hljs-comment"># -e &lt;ERROR RATE&gt;：最大允许错误率（错误数除以匹配区域的长度）默认值：0.1</span><br><span class="hljs-comment"># --gzip：使用 gzip 压缩输出文件。如果输入文件是 gzip 压缩的，则输出文件也将自动进行 gzip 压缩</span><br><span class="hljs-comment"># --length &lt;INT&gt;：丢弃由于质量或adapter修整而变得比长度INT短的读取。值为0可有效禁用此行为。默认值：20 bp。对于paired-end文件，读取配对的两个读数都需要长于bp才能打印到经过验证的配对末端文件（请参阅选项--paired）。如果只有一次读取变得太短，则可能保留此类未配对的单端读取（请参阅--retain_unpaired）。默认配对截断值：20 bp。</span><br><span class="hljs-comment"># -q/--quality &lt;INT&gt;：除了去除adapter外，还可以从读取中修剪低质量的末端。对于RRBS样品，首先进行质量修整，第二轮进行adapter修整。其他文件在一次通过中进行质量和adapter修剪。该算法与 BWA 使用的算法相同（从所有质量中减去 INT；计算从所有索引到序列末尾的部分和；在总和最小的索引处切割序列）。默认Phred分数：20</span><br></code></pre></td></tr></table></figure>
<h5 id="fastp"><a href="#fastp" class="headerlink" title="fastp"></a>fastp</h5><figure class="highlight apache"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><code class="hljs apache"><span class="hljs-attribute">nohup</span> fastp -i RM_CP_CTCF-ChIP-seq_<span class="hljs-number">1</span>.fastq.gz -I RM_CP_CTCF-ChIP-seq_<span class="hljs-number">2</span>.fastq.gz -o ../clean/RM_CP_CTCF_trim_R<span class="hljs-number">1</span>.fq.gz -O ../clean/RM_CP_CTCF_trim_R<span class="hljs-number">2</span>.fq.gz -w <span class="hljs-number">20</span> --detect_adapter_for_pe &amp;);done<br><span class="hljs-attribute">nohup</span> fastp -i RM_CP_CTCF-ChIP-seq_input_<span class="hljs-number">1</span>.fastq.gz -I RM_CP_CTCF-ChIP-seq_input_<span class="hljs-number">2</span>.fastq.gz -o ../clean/RM_CP_CTCF_input_trim_R<span class="hljs-number">1</span>.fq.gz -O ../clean/RM_CP_CTCF_input_trim_R<span class="hljs-number">2</span>.fq.gz -w <span class="hljs-number">20</span> --detect_adapter_for_pe &amp;);done<br></code></pre></td></tr></table></figure>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><code class="hljs python"><span class="hljs-comment">#再次质控</span><br>cd /sde/ckm/ChIP-seq/RM-CTCF/clean<br>ls *.gz |<span class="hljs-keyword">while</span> read <span class="hljs-built_in">id</span>; do (nohup fastqc -t <span class="hljs-number">20</span> -o ../qc_after $<span class="hljs-built_in">id</span> &amp; );done<br>multiqc ./<br><span class="hljs-comment">#过滤完，数据质量是ok的，但是过滤了一半的reads。。。</span><br><span class="hljs-comment">#也可以看到两个软件过滤的对比</span><br></code></pre></td></tr></table></figure>
<p><img src="image-20220210201437081.png" srcset="/img/loading.gif" lazyload></p>
<h4 id="4、比对bowtie2"><a href="#4、比对bowtie2" class="headerlink" title="4、比对bowtie2"></a>4、比对bowtie2</h4><figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><code class="hljs python"><span class="hljs-comment">#注意bowtie2不能对压缩的文件建索引，所以下载好的参考基因组必须要先解压</span><br>cd /sde/ckm/index/bowtie2/mmul_8<br>nohup bowtie2-build --threads <span class="hljs-number">10</span> ~/sde/ckm/reference/mmul_10/rheMac10.fa mmul10 &amp;<br><span class="hljs-comment">#结果</span><br>mmul10<span class="hljs-number">.1</span>.bt2 mmul10<span class="hljs-number">.2</span>.bt2 mmul10<span class="hljs-number">.3</span>.bt2 mmul10<span class="hljs-number">.4</span>.bt2 mmul10.rev<span class="hljs-number">.1</span>.bt2 mmul10.rev<span class="hljs-number">.2</span>.bt2<br><span class="hljs-comment">##############################################################################</span><br><span class="hljs-comment">#比对，生成的sam转bam，并且质控</span><br>cd /sde/ckm/ChIP-seq/RM-CTCF/align<br><span class="hljs-comment">#fastp</span><br>nohup bowtie2 -p <span class="hljs-number">20</span> --very-sensitive -x /sde/ckm/index/bowtie2/mmul_10/mmul10 -<span class="hljs-number">1</span> ../clean/RM_CP_CTCF_trim_R1.fq.gz -<span class="hljs-number">2</span> ../clean/RM_CP_CTCF_trim_R2.fq.gz | samtools view -@ <span class="hljs-number">20</span> -bS -q <span class="hljs-number">30</span> -o RM_CP_CTCF.MAPQ30.bam &amp;<br>nohup bowtie2 -p <span class="hljs-number">20</span> --very-sensitive -x /sde/ckm/index/bowtie2/mmul_10/mmul10 -<span class="hljs-number">1</span> ../clean/RM_CP_CTCF_input_trim_R1.fq.gz -<span class="hljs-number">2</span> ../clean/RM_CP_CTCF_input_trim_R2.fq.gz | samtools view -@ <span class="hljs-number">20</span> -bS -q <span class="hljs-number">30</span> -o RM_CP_CTCF_input.MAPQ30.bam &amp;<br><span class="hljs-comment">#trim-galore结果有问题，我得再看看文献啥的，查查为什么，所以后续均用fastp过滤的结果</span><br><span class="hljs-comment">#给bam排序</span><br>nohup samtools sort -@ <span class="hljs-number">20</span> RM_CP_CTCF.MAPQ30.bam -o RM_CP_CTCF.MAPQ30.sort.bam &amp;<br>nohup samtools sort -@ <span class="hljs-number">20</span> RM_CP_CTCF_input.MAPQ30.bam -o RM_CP_CTCF_input.MAPQ30.sort.bam &amp;<br></code></pre></td></tr></table></figure>
<h4 id="5、去除PCR重复"><a href="#5、去除PCR重复" class="headerlink" title="5、去除PCR重复"></a>5、去除PCR重复</h4><h5 id="picard"><a href="#picard" class="headerlink" title="picard"></a>picard</h5><figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br></pre></td><td class="code"><pre><code class="hljs python"><span class="hljs-comment">#picard</span><br>picard=/sde/ckm/miniconda3/envs/chipseq/share/picard-<span class="hljs-number">2.25</span><span class="hljs-number">.7</span>-<span class="hljs-number">0</span>/picard.jar<br><span class="hljs-comment">#测试picard是否安装了</span><br>java -jar $picard -h<br><span class="hljs-comment">#去除pcr重复（时间较长）</span><br>cd /sde/ckm/ChIP-seq/RM-CTCF/align<br>nohup java -jar $picard MarkDuplicates INPUT=RM_CP_CTCF.MAPQ30.sort.bam OUTPUT=p-RM_CP_CTCF.MAPQ30.sort.rmd.bam REMOVE_DUPLICATES=true METRICS_FILE=RM_CP_CTCF.marked_dup_metrics.txt &amp;<br>nohup java -jar $picard  MarkDuplicates INPUT=RM_CP_CTCF_input.MAPQ30.sort.bam OUTPUT=p-RM_CP_CTCF_input.MAPQ30.sort.rmd.bam REMOVE_DUPLICATES=true METRICS_FILE=RM_CP_CTCF_input.marked_dup_metrics.txt &amp;<br><br><span class="hljs-comment">#对比对质控</span><br>ls *p-* |xargs -i samtools index &#123;&#125; <span class="hljs-comment">#文件为.bai</span><br>samtools flagstat p-RM_CP_CTCF.MAPQ30.sort.rmd.bam&gt; p-RM_CP_CTCF.MAPQ30.sort.rmd.bam.stat<br>samtools flagstat p-RM_CP_CTCF_input.MAPQ30.sort.rmd.bam &gt; p-RM_CP_CTCF_input.MAPQ30.sort.rmd.bam.stat<br><span class="hljs-comment">#查看结果</span><br>cat *.stat |grep <span class="hljs-string">&quot;mapped (&quot;</span><br><span class="hljs-comment">#我都有点怀疑这个数据是不是真的，100%正常么</span><br><span class="hljs-number">27427512</span> + <span class="hljs-number">0</span> mapped (<span class="hljs-number">100.00</span>% : N/A)<br><span class="hljs-number">16521250</span> + <span class="hljs-number">0</span> mapped (<span class="hljs-number">100.00</span>% : N/A)<br></code></pre></td></tr></table></figure>
<h5 id="sambamba"><a href="#sambamba" class="headerlink" title="sambamba"></a>sambamba</h5><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><code class="hljs shell">nohup sambamba markdup -r -t 20 RM_CP_CTCF.MAPQ30.sort.bam RM_CP_CTCF.MAPQ30.sort.rmd.bam &amp;<br>nohup sambamba markdup -r -t 20 RM_CP_CTCF_input.MAPQ30.sort.bam RM_CP_CTCF_input.MAPQ30.sort.rmd.bam &amp;<br><span class="hljs-meta">#</span><span class="bash">-r, --remove-duplicates</span><br><span class="hljs-meta">#</span><span class="bash">-t, --nthreads</span><br></code></pre></td></tr></table></figure>
<h4 id="6、MACS2找peaks"><a href="#6、MACS2找peaks" class="headerlink" title="6、MACS2找peaks"></a>6、MACS2找peaks</h4><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br></pre></td><td class="code"><pre><code class="hljs shell"><span class="hljs-meta">#</span><span class="bash">picard</span><br>nohup macs2 callpeak -c p-RM_CP_CTCF_input.MAPQ30.sort.rmd.bam -t p-RM_CP_CTCF.MAPQ30.sort.rmd.bam -f BAM --nomodel --extsize 147 -g mm -n ../peaks/picard-RM_CP_CTCF -B &amp;<br><span class="hljs-meta">#</span><span class="bash">sambamba</span><br>nohup macs2 callpeak -c RM_CP_CTCF_input.MAPQ30.sort.rmd.bam -t RM_CP_CTCF.MAPQ30.sort.rmd.bam -f BAM --nomodel --extsize 147 -g mm -n ../peaks/sambamba-RM_CP_CTCF -B &amp;<br><span class="hljs-meta">#</span><span class="bash">macs2主要参数</span><br>-c：对照组<br>-t：实验组<br>-f：-t和-c的文件格式，如果不指定会自动检索<br>-B：更多信息保存到bedgraph中<br>-g：hs为human，mm为小鼠<br>-n：为输出文件的前缀名，可指定输出文件夹<br>-q：默认值为0.05，按此阈值来筛选peak<br><span class="hljs-meta">#</span><span class="bash"><span class="hljs-comment">#############################################################################</span></span><br><span class="hljs-meta">#</span><span class="bash">bed文件:一般代表区域信息，如表示Peak位置的bed文件，表示基因注释的bed文件。</span><br>bed文件一行表示一个gene，对应信息如下:<br>必须包含的3列信息：<br>1）chrom：染色体名字 (e.g.chr3, chrY, chr2_random或者scaffold10671)。<br>2）chromStart：基因在染色体或scaffold上的起始位置（0-based）。<br>3）chromEnd：基因在染色体或scaffold上的终止位置 （前闭后开）。<br>可选的9列信息：<br>4）name：bed文件的行名。<br>5）score：本条基因在注释数据集文件中的评分（0-1000），在Genome Browser中会根据不同区段的评分显示对应的阴影强度（评分越高灰度越高）。<br>6）strand：链的方向+、-或. (.表示不确定链的方向)<br>7）thickStart：CDS区（编码区）的起始位置，即起始密码子的位置。<br>8）thickEnd：The endingposition at which the feature is drawn thickly (for example the stop codon ingene displays).<br>9）itemRgb：RGB颜色值（如：255,0,0），方便在GenomeBrowser中查看。<br>10）blockCount：bed行中外显子的数目。<br>11）blockSizes：逗号分割的列，数目与blockCount值对应，每个数表示对应外显子的碱基数。<br>12）blockStarts：逗号分割的列，数目与blockCount值对应，每个数表示对应外显子的起始位置（数值是相对ChromStart计算的）<br><span class="hljs-meta">#</span><span class="bash">bdg、bw文件</span><br>导入基因组浏览器查看比对状态和突变信息。wiggle(简称wig)、bigwig(简写bw)以及bedgraph(简写bdg)只包含区域和区域的覆盖度信息，文件更小，用于可视化更方便，可以导入基因组浏览器（Genome Browser）中进行可视化，以查看reads在参考基因组各个区域的覆盖度并检测测序深度。<br>wiggle：展示区域的密度或强度信息，如GCpercent, probability scores, and transcriptome data.<br>bedGraph: bed与wig的结合，更省空间和灵活，展示信息与wig类似。 (bedGraph的格式一般有四列，Chr、start、end和value，并且坐标是以0为起始左闭右开)<br>bigWig: wig文件的二进制压缩格式，可通过wigToBigWig工具转换<br><span class="hljs-meta">#</span><span class="bash"><span class="hljs-comment">#############################################################################</span></span><br><span class="hljs-meta">#</span><span class="bash">结果</span><br>wc -l picard-RM_CP_CTCF_summits.bed<br>33671 picard-RM_CP_CTCF_summits.bed<br><span class="hljs-meta">#</span><span class="bash">结果</span><br>wc -l fastp-RM_CP_CTCF_summits.bed<br>33671 fastp-RM_CP_CTCF_summits.bed<br><span class="hljs-meta">#</span><span class="bash">picard结果和sambamba结果一毛一样</span><br></code></pre></td></tr></table></figure>
<p><img src="image-20220210220849746.png" srcset="/img/loading.gif" lazyload></p>
<p><img src="image-20220210221139367.png" srcset="/img/loading.gif" lazyload></p>

            </div>
            <hr>
            <div>
              <div class="post-metas mb-3">
                
                
                  <div class="post-meta">
                    <i class="iconfont icon-tags"></i>
                    
                      <a class="hover-with-bg" href="/tags/%E7%BB%84%E5%AD%A6%E5%88%86%E6%9E%90/">组学分析</a>
                    
                      <a class="hover-with-bg" href="/tags/ChIP-seq/">ChIP-seq</a>
                    
                  </div>
                
              </div>
              
                <p class="note note-warning">
                  
                    本博客所有文章除特别声明外，均采用 <a target="_blank" href="https://creativecommons.org/licenses/by-sa/4.0/deed.zh" rel="nofollow noopener noopener">CC BY-SA 4.0 协议</a> ，转载请注明出处！
                  
                </p>
              
              
                <div class="post-prevnext">
                  <article class="post-prev col-6">
                    
                    
                  </article>
                  <article class="post-next col-6">
                    
                    
                      <a href="/2022/01/29/%E8%BF%90%E8%A1%8CHicPro%E5%87%BA%E7%8E%B0mergeSAM%E6%8A%A5%E9%94%99%E5%8F%8A%E8%A7%A3%E5%86%B3%E8%BF%87%E7%A8%8B/">
                        <span class="hidden-mobile">运行HicPro出现Makefile:141:recipe_for_target'bowtie_pairing'failed报错及解决过程</span>
                        <span class="visible-mobile">下一篇</span>
                        <i class="iconfont icon-arrowright"></i>
                      </a>
                    
                  </article>
                </div>
              
            </div>

            
          </article>
        </div>
      </div>
    </div>
    
      <div class="d-none d-lg-block col-lg-2 toc-container" id="toc-ctn">
        <div id="toc">
  <p class="toc-header"><i class="iconfont icon-list"></i>&nbsp;目录</p>
  <div class="toc-body" id="toc-body"></div>
</div>

      </div>
    
  </div>
</div>

<!-- Custom -->


    

    
      <a id="scroll-top-button" aria-label="TOP" href="#" role="button">
        <i class="iconfont icon-arrowup" aria-hidden="true"></i>
      </a>
    

    
      <div class="modal fade" id="modalSearch" tabindex="-1" role="dialog" aria-labelledby="ModalLabel"
     aria-hidden="true">
  <div class="modal-dialog modal-dialog-scrollable modal-lg" role="document">
    <div class="modal-content">
      <div class="modal-header text-center">
        <h4 class="modal-title w-100 font-weight-bold">搜索</h4>
        <button type="button" id="local-search-close" class="close" data-dismiss="modal" aria-label="Close">
          <span aria-hidden="true">&times;</span>
        </button>
      </div>
      <div class="modal-body mx-3">
        <div class="md-form mb-5">
          <input type="text" id="local-search-input" class="form-control validate">
          <label data-error="x" data-success="v"
                 for="local-search-input">关键词</label>
        </div>
        <div class="list-group" id="local-search-result"></div>
      </div>
    </div>
  </div>
</div>
    

    
  </main>

  <footer class="text-center mt-5 py-3">
  <div class="footer-content">
     <a href="https://hexo.io" target="_blank" rel="nofollow noopener"><span>Hexo</span></a> <i class="iconfont icon-love"></i> <a href="https://github.com/fluid-dev/hexo-theme-fluid" target="_blank" rel="nofollow noopener"><span>Fluid</span></a> 
  </div>
  
  <div class="statistics">
    
    

    
      
        <!-- LeanCloud 统计PV -->
        <span id="leancloud-site-pv-container" style="display: none">
            总访问量 
            <span id="leancloud-site-pv"></span>
             次
          </span>
      
      
        <!-- LeanCloud 统计UV -->
        <span id="leancloud-site-uv-container" style="display: none">
            总访客数 
            <span id="leancloud-site-uv"></span>
             人
          </span>
      

    
  </div>


  

  
</footer>


  <!-- SCRIPTS -->
  
  <script  src="https://cdn.jsdelivr.net/npm/nprogress@0/nprogress.min.js" ></script>
  <link  rel="stylesheet" href="https://cdn.jsdelivr.net/npm/nprogress@0/nprogress.min.css" />

  <script>
    NProgress.configure({"showSpinner":false,"trickleSpeed":100})
    NProgress.start()
    window.addEventListener('load', function() {
      NProgress.done();
    })
  </script>


<script  src="https://cdn.jsdelivr.net/npm/jquery@3/dist/jquery.min.js" ></script>
<script  src="https://cdn.jsdelivr.net/npm/bootstrap@4/dist/js/bootstrap.min.js" ></script>
<script  src="/js/events.js" ></script>
<script  src="/js/plugins.js" ></script>

<!-- Plugins -->


  <script  src="/js/local-search.js" ></script>



  
    <script  src="/js/img-lazyload.js" ></script>
  



  



  
    <script  src="https://cdn.jsdelivr.net/npm/tocbot@4/dist/tocbot.min.js" ></script>
  
  
    <script  src="https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@3/dist/jquery.fancybox.min.js" ></script>
  
  
    <script  src="https://cdn.jsdelivr.net/npm/anchor-js@4/anchor.min.js" ></script>
  
  
    <script defer src="https://cdn.jsdelivr.net/npm/clipboard@2/dist/clipboard.min.js" ></script>
  




  <script defer src="/js/leancloud.js" ></script>



  <script  src="https://cdn.jsdelivr.net/npm/typed.js@2/lib/typed.min.js" ></script>
  <script>
    (function (window, document) {
      var typing = Fluid.plugins.typing;
      var title = document.getElementById('subtitle').title;
      
        typing(title);
      
    })(window, document);
  </script>












  

  

  

  

  

  





<!-- 主题的启动项 保持在最底部 -->
<script  src="/js/boot.js" ></script>


</body>
</html>
